tools/generate-models/run.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env zsh
+
+MODE_VERIFY="verify"
+MODE_GENERATE="generate"
+MODE="$1" # verify or generate
+TARGET_FILE="$(pwd)/$2"
+
+# get git reference
+if [[ $MODE == $MODE_VERIFY ]]; then
+    TARGET_LINE=$(tail -n 1 $TARGET_FILE)
+    SHA_REGEX='[0-9a-f]{5,40}'
+    if [[ $TARGET_LINE =~ $SHA_REGEX ]]; 
+    then
+        GIT_REF=$MATCH
+    else
+        echo "Target SHA could not be read in $TARGET_FILE !"
+        return 1;
+    fi
+elif [[ $MODE == $MODE_GENERATE ]]; then
+    GIT_REF="master"
+else
+    echo "Invalid command!"
+    echo "Model generation: run.sh generate ${TARGET_LOCATION_FOR_NEW_FILE}"
+    echo "Model verification: run.sh verify ${FILE_TO_BE_VERIFIED}"
+    return 1;
+fi
+
+echo "Git ref:"
+echo $GIT_REF
+
+# fetch rum-events-format for given git ref
+SCRIPT_LOC=$(dirname "$0")
+cd $SCRIPT_LOC
+if [[ ! -d "rum-events-format" ]]; then
+    git clone git@github.com:DataDog/rum-events-format.git
+fi
+cd "rum-events-format"
+git fetch origin $GIT_REF
+git checkout FETCH_HEAD
+SHA=$(git rev-parse HEAD)
+cd ..
+
+# install quicktype and generate model
+if [[ ! -a $(npm bin)/quicktype ]]; then
+    npm install buranmert/quicktype
+fi
+$(npm bin)/quicktype --lang swift --src-lang schema -o RUMModels.swift.bak --type-prefix RUM --src rum-events-format/schemas
+echo "// $SHA" >> RUMModels.swift.bak
+
+# verify or move the generated model
+if [[ $MODE == $MODE_VERIFY ]]; then
+    DIFF=$(diff RUMModels.swift.bak $TARGET_FILE)
+    if [[ ! -z $DIFF ]]; then 
+        echo "Diff detected!\n$DIFF"
+        return 1;
+    fi
+    echo "Verified!"
+    return 0;
+elif [[ $MODE == $MODE_GENERATE ]]; then
+    cp RUMModels.swift.bak $TARGET_FILE
```

Comment:
As we expose `protocol RUMDataModel`, let's name the target file `RUMDataModels.swift`, WDYT?